app/controllers/cohorts_controller.rb

```
@@ -14,10 +15,11 @@ def show
   end
 
   def my
-    if current_user && current_user.student?
-      redirect_to cohort_path(current_user.student.cohort)
+    if current_user
+      redirect_to cohort_path(current_user.student.cohort) if current_user.student?
+      redirect_to staff_cohorts_path if current_user.instructor?
     else
-      render "Not enrolled"
+      redirect_to '/404', alert: "Not enrolled in a cohort. Contact your instructor."
```

Actually, we should raise a page not found exception.